### PR TITLE
[corechecks/containerlifecycle] Add Pod finished timestamp to containerlifecycle pod events

### PR DESCRIPTION
**Pull request by sblumenthal**
***NOTE: this PR is waiting on others to be merged in first, so some bits may change***

### What does this PR do?

This PR adds a pod exit timestamp to the PodEvents emitted by the container-lifecycle check, which is defined [here](https://github.com/DataDog/agent-payload/pull/240)

### Motivation

Currently, we have the timestamp when a container terminates, but we do not have it on the pod side.

### Additional Notes

This PR is waiting on the linked agent-payload PR to be merged, which itself is waiting on another agent-payload PR to be merged, which might affect the underlying data model used in this PR. As a result, some syntax might change between now and when the PR is marked as ready.

### Possible Drawbacks / Trade-offs

We do not actually get the timestamp that a pod is killed by the kubelet, so to get around this I am using the timestamp of the first workloadmeta pull in which we did not see the pod anymore. The implementation currently sets all pods that were removed between intervals to have the same exited timestamp.

### Describe how to test/QA your changes

Testing this is tricky, because (as far as I know) we do not have a way to inspect the payload generated by this check. I was able to validate this locally by building the agent with debug flags set, and then using delve to set breakpoints at the return statements of `toPayloadModel` and `toEventModel` in `pkg/collector/corechecks/containerlifecycle/event.go`. I was able to validate locally that the field is populated this way.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -40,7 +40,7 @@ require (
 	code.cloudfoundry.org/garden v0.0.0-20210208153517-580cadd489d2
 	code.cloudfoundry.org/lager v2.0.0+incompatible
 	github.com/CycloneDX/cyclonedx-go v0.7.0
-	github.com/DataDog/agent-payload/v5 v5.0.76
+	github.com/DataDog/agent-payload/v5 v5.0.77-0.20230316174929-5682772a5eaa
```
Comment (sblumenthal) [inline go.mod:43]: TODO: update this after https://github.com/DataDog/agent-payload/pull/240 is merged in

File: pkg/workloadmeta/types.go [hunk 2]
```diff
@@ -531,6 +532,7 @@ func (p KubernetesPod) String(verbose bool) string {
 		_, _ = fmt.Fprintln(&sb, "PVCs:", sliceToString(p.PersistentVolumeClaimNames))
 		_, _ = fmt.Fprintln(&sb, "Kube Services:", sliceToString(p.KubeServices))
 		_, _ = fmt.Fprintln(&sb, "Namespace Labels:", mapToString(p.NamespaceLabels))
+		_, _ = fmt.Fprintln(&sb, "Finished At:", p.FinishedAt)
```
Comment (davidor) [inline pkg/workloadmeta/types.go:535]: Should we print this only if it's non-zero?

Comment (sblumenthal) [reply pkg/workloadmeta/types.go:535]: I think that makes sense. I'll make the change!

**Review by maycmlee — APPROVED** [review]
👍 for docs

**Review by vboulineau — APPROVED** [review]

**Pull request merged**